src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,93 @@
+import classNames from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  items: string[];
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  items,
+  onPageChange,
+}) => {
+  const totalPage = Math.ceil(total / perPage);
+  const totalPageArray = [];
+
+  for (let i = 1; i < totalPage + 1; i++) {
+    totalPageArray.push(i);
+  }
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={classNames('page-item', {
+            'page-item disabled': currentPage === 1,
+          })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 1}
+            onClick={() => {
+              if (currentPage > 1) {
+                onPageChange(currentPage - 1);
+              }
+            }}
+          >
+            «
+          </a>
+        </li>
+        {totalPageArray.map(page => (
+          <li
+            key={page}
+            className={classNames('page-item', {
+              'page-item active': currentPage === page,
+            })}
+          >
+            <a
+              data-cy="pageLink"
+              className="page-link"
+              href="#2"
+              onClick={() => onPageChange(page)}
+            >
+              {page}
+            </a>
+          </li>
+        ))}
+        <li
+          className={classNames('page-item', {
+            'page-item disabled': currentPage === totalPage,
+          })}
+        >
+          <a
+            data-cy="nextLink"
+            className="page-link"
+            href="#next"
+            aria-disabled={currentPage === totalPage}
+            onClick={() => {
+              if (currentPage < totalPage) {
+                onPageChange(currentPage + 1);
+              }
+            }}
+          >
+            »
+          </a>
+        </li>
+      </ul>
+      <ul>
+        {items.map(item => (
+          <li key={item} data-cy="item">
+            {item}
+          </li>
+        ))}
+      </ul>
```

Comment:
Rendering items within the `Pagination` component is unconventional. It is better to handle item rendering in the parent component to maintain separation of concerns and make the `Pagination` component more reusable.